custom-stats-logger/src/main/java/dev/gemfire/stat/logger/StatLogger.java

```diff
@@ -0,0 +1,151 @@
+  private Timer timer;
+  private TimerTask task;
+  private long timerInterval = 5000;
+
+  @Override
+  public void initialize(Cache cache, Properties props) {
+    logger.info("Initializing Stat Logger with properties {}", props);
+  }
+
+  @Override
+  public boolean isHA() {
+    return false;
+  }
+
+  @Override
+  public boolean hasResult() {
+    return true;
+  }
+
+  @Override
+  public String getId() {
+    return StatLogger.class.getSimpleName();
+  }
+
+  @Override
+  public void execute(FunctionContext functionContext) {
+    ClassLoader contextClassLoader = this.getClass().getClassLoader();
+    Properties properties = new Properties();
+    try {
+      final InputStream stream = contextClassLoader.getResourceAsStream("statLogger.properties");
+      properties.load(stream);
+    } catch (IOException e) {
+      logger.error("Could not load statLogger.properties", e);
+    }
+    String logFrequency = properties.getProperty("logFrequency");
+    String statistics = properties.getProperty("statistics");
+
+    cleanup(functionContext.getCache());
+
+    if (!StringUtils.isEmpty(statistics) && !StringUtils.isEmpty(logFrequency)) {
+      createTimerRegion(functionContext.getCache());
+
+      timerInterval = Long.parseLong(logFrequency);
+      createStatsList(functionContext.getCache().getDistributedSystem(), statistics);
+      ensureTimerRunning(functionContext.getCache());
+      functionContext.getResultSender().lastResult("Logging Metric count " + listOfStats.size() +
+              " with timer interval set to " + timerInterval + " ms");
+
+    } else if (StringUtils.isEmpty(statistics) && !StringUtils.isEmpty(logFrequency)) {
+      functionContext.getResultSender().lastResult("No stats to log. StatLogger will not start.");
+
+    } else if (!StringUtils.isEmpty(statistics) && StringUtils.isEmpty(logFrequency)) {
+      createTimerRegion(functionContext.getCache());
+
+      createStatsList(functionContext.getCache().getDistributedSystem(), statistics);
+      ensureTimerRunning(functionContext.getCache());
+      functionContext.getResultSender().lastResult("Logging Metric count " + listOfStats.size() +
+              " with timer interval set to default of " + timerInterval + " ms");
+
+    } else {
+      functionContext.getResultSender().lastResult("No properties set for StatLogger. StatLogger will not start.");
+
+    }
+  }
+
+  private void createStatsList(DistributedSystem distributedSystem, String statistics) {
+    List<String> statisticsList = Arrays.asList(statistics.split(",", 0));
+    for (String statistic : statisticsList) {
+      String[] statsNames = statistic.split("[.]", 0);
+      addStats(distributedSystem, statsNames[0], statsNames[1]);
+    }
+  }
+
+  private static void createTimerRegion(Cache cache) {
+    RegionFactory<Object, Object> regionFactory = cache.createRegionFactory(RegionShortcut.LOCAL);
+    regionFactory.create("Timer");
```

Comment:
should the region name be more unique to make sure it does not interfere with a customer's region with the same name?
For example "GemFireStatLoggerTimer"